File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.py
def key_result(keymap, target):
    re = 0
    
    for char in target:
        
        num = 101
        temp = 102
        for key in keymap:
            if char in key:
                temp = key.index(char) + 1
            if num > temp:
                num = temp
        if (temp == 102):
            return (-1)
        re += num
        
        
    return (re)

def solution(keymap, targets):
    answer = []
    for tarr in targets:
        answer.append(key_result(keymap, tarr))
    
    return answer

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <iostream>

using namespace std;

vector<int> solution(vector<string> keymap, vector<string> targets) {
    vector<int> answer;
    unordered_map <char, int> key;
    
    for (const auto &keys : keymap) {
        for (int i = 0; i < keys.size(); i++) {
            if (key.find(keys[i]) != key.end()) key[keys[i]] = min(key[keys[i]], i + 1);
            else key[keys[i]] = i + 1;
        }
    }
    

    
    for (const auto &target : targets) {
        int cnt = 0;
        for (const auto &charset : target) {
            if (key.find(charset) == key.end()) {
                cnt = -1;
                break;
            }
            cnt += key[charset];
        }
        answer.push_back(cnt);
    }
    return answer;
}